components/profile.js

```diff
@@ -65,57 +67,413 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
+										}
+										out.push({
+											"name": "rgShowcaseConfig[17][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[17][" + n + "][title]",
+											"value": defaultval[1]
+										});
+									}
+								}
+								break;
+
+								case 'games':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "2"
+									});
+
+									if (settings[i][t]["values"].hasOwnProperty("games")) {
+										showcaseconfig = [true, 4, 2, settings[i][t]["values"]["games"]];
+									}
+									break;
+
+								case 'ownguides':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "16"
+									});
+
+									for (var n = 0; n < 4; n++) {
+										var defaultval = ["", ""];
+										if (settings[i][t]["values"][n] != undefined) {
+											defaultval = [settings[i][t]["values"][n]["appid"], settings[i][t]["values"][n]["publishedfileid"]];
+										}
+										out.push({
+											"name": "rgShowcaseConfig[16][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[16][" + n + "][publishedfileid]",
+											"value": defaultval[1]
+										});
+									}
+									break;
+
+								case 'ownguides':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "12"
+									});
+
+									for (var n = 0; n < 5; n++) {
+										var defaultval = ["", ""];
+										if (settings[i][t]["values"][n] != undefined) {
+											defaultval = [settings[i][t]["values"][n]["appid"], settings[i][t]["values"][n]["publishedfileid"]];
+										}
+										out.push({
+											"name": "rgShowcaseConfig[12][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[12][" + n + "][publishedfileid]",
+											"value": defaultval[1]
+										});
+									}
+									break;
+							}
+
+							if (showcaseconfig[0]) {
+								for (var n = 0; n < showcaseconfig[1]; n++) {
+									var requestdata;
+									if (showcaseconfig[3][n] == undefined) {
+										requestdata = {
+											appid: 0,
+											item_contextid: 0,
+											item_assetid: 0,
+											customization_type: showcaseconfig[2],
+											slot: n,
+											sessionid: formd[0].value
+										};
+									} else {
+										requestdata = {
+											appid: showcaseconfig[3][n]["appid"],
+											item_contextid: showcaseconfig[3][n]["item_contextid"],
+											item_assetid: showcaseconfig[3][n]["item_assetid"],
+											customization_type: showcaseconfig[2],
+											slot: n,
+											sessionid: formd[0].value
+										};
+
+									}
+
+									self._myProfile("ajaxsetshowcaseconfig", requestdata, function (err, response, body) {
```

Comment:
Are we making HTTP requests in a loop? IIRC there's no internal throttling, so this is begging for rate limit.

Reply:
I can add like 1 second delay with `setTimeout` should it be really needed. But I tested if it was a problem for the endpoint if you made multiple request's at once, and didn't get an error.